clar2wasm/src/wasm_generator.rs

```diff
@@ -3048,19 +3066,56 @@ impl WasmGenerator {
         for operand in operands {
             self.traverse_expr(builder, operand)?;
         }
+
         self.visit_comparison(builder, expr, func, operands)
     }
 
-    fn traverse_logical(
+    fn traverse_comparison_equals(
+        &mut self,
+        builder: &mut InstrSeqBuilder,
+        expr: &SymbolicExpression,
+        func: NativeFunctions,
+        operands: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        let helper_func = match func {
+            NativeFunctions::Equals => self
+                .module
+                .funcs
+                .by_name("is-eq")
+                .unwrap_or_else(|| panic!("function not found: is-eq")),
+            _ => {
+                return Err(GeneratorError::NotImplemented);
+            }
+        };
+
+        self.traverse_expr(builder, &operands[0])?;
+        for operand in operands.iter().skip(1) {
+            self.traverse_expr(builder, operand)?;
+            builder.call(helper_func);
+        }
```

Comment:
Hey @obycode ! This is an initial implementation for the `is-eq` expression. It seems to be working with the amount of operators required (considering all operators in the function. Not a short-circuit).

But there is an issue when running the function for more than two operands. Inside the `for` loop, the result is always a  boolean i32 that is passed to the webassembly function, but there is expecting a i64. For instance:

```
(is-eq 1 2 3)
(is-eq (is-eq 1 2) 3) --> (is-eq (is-eq i64 i64) i64) --> (is-eq i32 i64)  
```

Any clue on that? Thanks

Reply:
Your `is-eq` function returns an `i32`, indicating true or false if the two arguments are equal, so you can't directly pass that result into `is-eq` with the next operand. The interpreter's implementation compares each operand with the first. That might be a good solution here as well.

Reply:
So instead of the transformation you described above, which is what we want for operators like `+`, for `is-eq` we instead would transform it like:
```
(and (is-eq 1 2) (is-eq 1 3))
```
Except it can't be exactly that, because `and` short-circuits and `is-eq` does not, but logically the same.